code_benches/run.py

```diff
@@ -27,32 +44,59 @@
 
 RUNTIME_PATH = ROOT_PATH + "/runtime"
 
-SILVERFISH_RELEASE_PATH = ROOT_PATH + "/target/release/silverfish"
-SILVERFISH_DEBUG_PATH = ROOT_PATH + "/target/debug/silverfish"
-assert all(arg in {"--release", "--debug"} for arg in sys.argv[1:])
-if "--release" in sys.argv:
-    SILVERFISH_PATH = SILVERFISH_RELEASE_PATH
-elif "--debug" in sys.argv:
-    SILVERFISH_PATH = SILVERFISH_DEBUG_PATH
-else:
-    def getmtime_or_zero(path):
+def bestpath(paths):
+    """
+    Determine best path based on:
+    1. exists
+    2. most recently modified
+    """
+    def getmtime_or_zero(x):
+        path, *_, use_this_one = x
+
+        if use_this_one:
+            return (1, 0)
+
         try:
-            return os.path.getmtime(path)
+            return (0, os.path.getmtime(path))
         except FileNotFoundError:
-            return 0
-    SILVERFISH_PATH = max(
-        [SILVERFISH_RELEASE_PATH, SILVERFISH_DEBUG_PATH],
-        key=getmtime_or_zero)
+            return (0, 0)
+
+    *best, _ = max(paths, key=getmtime_or_zero)
+    return best
+
+SILVERFISH_RELEASE_PATH = ROOT_PATH + "/target/release/silverfish"
+SILVERFISH_DEBUG_PATH = ROOT_PATH + "/target/debug/silverfish"
+
+SILVERFISH_PATH, = bestpath([
+    (SILVERFISH_RELEASE_PATH, args.release),
+    (SILVERFISH_DEBUG_PATH,   args.debug),
+])
 
 WASMCEPTION_PATH = ROOT_PATH + "/wasmception"
+WASMCEPTION_CLANG = WASMCEPTION_PATH + "/dist/bin/clang"
+WASMCEPTION_SYSROOT = WASMCEPTION_PATH + "/sysroot"
+WASMCEPTION_FLAGS = "--target=wasm32-unknown-unknown-wasm -nostartfiles -O3 -flto"
+WASMCEPTION_BACKING = "wasmception_backing.c"
+
+WASI_SDK_PATH = ROOT_PATH + "/wasi-sdk"
+WASI_SDK_CLANG = WASI_SDK_PATH + "/bin/clang"
+WASI_SDK_SYSROOT = WASI_SDK_PATH + "/share/wasi-sysroot"
+WASI_SDK_FLAGS = "--target=wasm32-wasi -mcpu=mvp -nostartfiles -O3 -flto"
```

Comment:
are there other options for cpu?

Reply:
Ah! At the time of writing this there are three:
- mvp (minimal viable product)
- bleeding-edge
- generic

Here:
https://github.com/llvm/llvm-project/blob/8c3e6af71b8e827655c83997747a2042feddc845/clang/lib/Basic/Targets/WebAssembly.cpp#L33-L34

I chose mvp as I know aWsm currently is inexaustive. I don't think there's actually a difference between mvp and generic at the moment, but there's a lot of proposed movement on Wasm that may cause issues.

A handful of non-trapping float instructions (fairly inconsequential) were added to the default cpu here:
https://reviews.llvm.org/D77908

There's also the recently added multi-valued return values that got added to Wasm recently. This may or may not be adopted in the default CPU. Related issue:
https://github.com/WebAssembly/wasi-sdk/issues/122

If someone wants to go through and make sure Wasm is covered exhaustively we could adopt -mcpu=default, but I worry this is a moving target.

Reply:
👍 something to think about in the future